tests/ComTests.cpp

```diff
@@ -3058,4 +3060,175 @@ TEST_CASE("COMEnumerator", "[com][enumerator]")
 #pragma warning(pop)
 #endif // WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP) && WIL_HAS_CXX_17
 
+#if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP | WINAPI_PARTITION_SYSTEM)
+#if (NTDDI_VERSION >= NTDDI_WINBLUE)
+#if defined(__cpp_impl_coroutine) || defined(__cpp_coroutines) || defined(_RESUMABLE_FUNCTIONS_SUPPORTED)
+
+#include <wil/cppwinrt_register_com_server.h>
+#include <winrt/windows.foundation.h>
+#include <windows.foundation.h>
+
+const winrt::guid CLSID_RPCTimeoutTestServer{L"CED2F47C-200B-476F-BFDC-D0D79B052AC6"};
+HANDLE g_hangHandle{nullptr};
+HANDLE g_doneHangingHandle{nullptr};
+
+TEST_CASE("com_timeout", "[com][com_timeout]")
+{
+    auto init = wil::CoInitializeEx_failfast();
+
+    // These test cases require a COM server that will use RPC for communication so that we can exercise
+    // RPC cancellation.  Additionally, this server needs to support the ability to control if it hangs

```

Comment:
Inproc calls don't go through RPC. COM does use NDR (implemented by rpcrt4.dll) to translate the call into request/response messages. But NDR is generally considered distinct from the RPC runtime. And in the inproc case, we don't use the RPC runtime to transport messages between client and server.

Consider "These test cases require calling a COM server via proxy, so that we can exercise cancellation through the COM runtime."

Normally wouldn't quibble, but comments in public testcases are a bit documentationy.

Reply:
Copied the suggestion verbatim.